### ISSUE TITLE
400 The computed checksum does not match

### ISSUE THREAD

**Issue by Crush33** [owncloud/core#31235]
Hi
I got a very very nasty issue right now on one of my owncloud server.
Every client (windows 10 mostly running client v2.4.1) got a "Connection Closed" after trying to upload any file bigger than 1MB.

Owncloud is running on Debian 9 virtualized on winserv 2016 and everything was running fine until my last apt update / upgrade a few days ago.

The error in owncloud.log is : "400 The computed checksum does not match the one received from the client." And i really don't know why this is happening...

Note : i run a "shrink" operation on my owncloud virtual hard disk which is running in HyperV which i can explain more in details if necessary.

### Steps to reproduce
1. inside a folder, put a file bigger than 1 MB
2. the upload is beginning then seems to stop then retry from zero (very quickly) then upload again

### Expected behaviour
Upload correctly the files

### Actual behaviour
after a few sec "Connection Closed" message appears on the client and upload fail.

### Server configuration
**Operating system**: 
Debian 9 virtualized on winserv 2016

**Web server:** 
Apache2.4

**Database:**
MariaDB 10.1.26-MariaDB-0+deb9u1 Debian 9.1

**PHP version:**
PHP7

**ownCloud version:** (see ownCloud admin page)
10.0.7.2

**Updated from an older ownCloud or fresh install:**
from 9.0

**Where did you install ownCloud from:**
net installer

**Signing status (ownCloud 9.0 and above):**

```
No errors have been found.
```

**The content of config/config.php:**

```
{
    "system": {
        "instanceid": "oc3w5y242iyo",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "192.168.1.12"
        ],
        "datadirectory": "\/var\/www\/owncloud\/data",
        "overwrite.cli.url": "https:\/\/cloud1.nodsys.com",
        "dbtype": "mysql",
        "version": "10.0.7.2",
        "dbname": "owncloud",
        "dbhost": "localhost",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "logtimezone": "UTC",
        "installed": true,
        "overwritecondaddr": "cloud1.nodsys.com",
        "overwritehost": "cloud1.nodsys.com",
        "mail_smtpmode": "smtp",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "465",
        "mail_smtpauthtype": "LOGIN",
        "mail_smtpauth": 1,
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpsecure": "ssl",
        "memcache.local": "\\OC\\Memcache\\APCu",
        "log_rotate_size": 26214400,
        "maintenance": false,
        "theme": "nodsys-theme",
        "loglevel": 2,
        "singleuser": false
    }
}
```

**List of activated apps:**

```
Enabled:
  - activity: 2.3.6
  - calendar: 1.5.7
  - comments: 0.3.0
  - configreport: 0.1.1
  - contacts: 1.5.4
  - dav: 0.3.2
  - federatedfilesharing: 0.3.1
  - federation: 0.1.0
  - files: 1.5.1
  - files_external: 0.7.1
  - files_sharing: 0.10.1
  - files_trashbin: 0.9.1
  - files_versions: 1.3.0
  - files_videoplayer: 0.9.8
  - firstrunwizard: 1.1
  - market: 0.2.3
  - notifications: 0.3.2
  - provisioning_api: 0.5.0
  - systemtags: 0.3.0
  - templateeditor: 0.2
  - updatenotification: 0.2.1
Disabled:
  - encryption
  - external
  - theme-example
  - user_external
```

**Are you using external storage, if yes which one:** local/smb/sftp/...
no

**Are you using encryption:** yes/no
no

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
no

### Client configuration
**Browser:**
chrome

**Operating system:**
win10

#### Web server error log
```
nothin ! no error at all
```

#### ownCloud log (data/owncloud.log)
```
{"reqId":"cjNnvqR4ZmzcRkwZdxwb","level":4,"time":"2018-04-21T13:20:24+00:00","remoteAddr":"193.250.180.152","user":"mga","app":"webdav","method":"PUT","url":"\/remote.php\/dav\/files\/mga\/MGA\/FACTURATION\/tableau%20facturation.xls","message":"Exception: {\"Message\":\"HTTP\\\/1.1 423 \\\"MGA\\\/FACTURATION\\\/tableau facturation.xls\\\" is locked\",\"Exception\":\"OCA\\\\DAV\\\\Connector\\\\Sabre\\\\Exception\\\\FileLocked\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/owncloud\\\/lib\\\/public\\\/Events\\\/EventEmitterTrait.php(50): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\File->OCA\\\\DAV\\\\Connector\\\\Sabre\\\\{closure}()\\n#1 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/lib\\\/Connector\\\/Sabre\\\/File.php(269): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\File->emittingCall(Object(Closure), Array, 'file', 'create')\\n#2 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(1129): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\File->put(Resource id #84)\\n#3 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/CorePlugin.php(513): Sabre\\\\DAV\\\\Server->updateFile('files\\\/mga\\\/MGA\\\/F...', Resource id #84, NULL)\\n#4 [internal function]: Sabre\\\\DAV\\\\CorePlugin->httpPut(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#5 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n#6 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(479): Sabre\\\\Event\\\\EventEmitter->emit('method:PUT', Array)\\n#7 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(254): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#8 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/lib\\\/Server.php(263): Sabre\\\\DAV\\\\Server->exec()\\n#9 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/appinfo\\\/v2\\\/remote.php(31): OCA\\\\DAV\\\\Server->exec()\\n#10 \\\/var\\\/www\\\/owncloud\\\/remote.php(165): require_once('\\\/var\\\/www\\\/ownclo...')\\n#11 {main}\",\"File\":\"\\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/lib\\\/Connector\\\/Sabre\\\/File.php\",\"Line\":205,\"User\":\"mga\"}"}
{"reqId":"DkzwV88R4apFlsqe9U2O","level":4,"time":"2018-04-21T13:20:50+00:00","remoteAddr":"193.250.180.152","user":"mga","app":"webdav","method":"PUT","url":"\/remote.php\/dav\/files\/mga\/MGA\/FACTURATION\/tableau%20facturation.xls","message":"Exception: {\"Message\":\"HTTP\\\/1.1 400 The computed checksum does not match the one received from the client.\",\"Exception\":\"Sabre\\\\DAV\\\\Exception\\\\BadRequest\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/owncloud\\\/lib\\\/public\\\/Events\\\/EventEmitterTrait.php(50): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\File->OCA\\\\DAV\\\\Connector\\\\Sabre\\\\{closure}()\\n#1 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/lib\\\/Connector\\\/Sabre\\\/File.php(269): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\File->emittingCall(Object(Closure), Array, 'file', 'create')\\n#2 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(1129): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\File->put(Resource id #84)\\n#3 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/CorePlugin.php(513): Sabre\\\\DAV\\\\Server->updateFile('files\\\/mga\\\/MGA\\\/F...', Resource id #84, NULL)\\n#4 [internal function]: Sabre\\\\DAV\\\\CorePlugin->httpPut(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#5 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n#6 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(479): Sabre\\\\Event\\\\EventEmitter->emit('method:PUT', Array)\\n#7 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(254): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#8 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/lib\\\/Server.php(263): Sabre\\\\DAV\\\\Server->exec()\\n#9 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/appinfo\\\/v2\\\/remote.php(31): OCA\\\\DAV\\\\Server->exec()\\n#10 \\\/var\\\/www\\\/owncloud\\\/remote.php(165): require_once('\\\/var\\\/www\\\/ownclo...')\\n#11 {main}\",\"File\":\"\\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/lib\\\/Connector\\\/Sabre\\\/File.php\",\"Line\":163,\"User\":\"mga\"}"}
{"reqId":"B1KOYhPUaB5XQcApAMO5","level":4,"time":"2018-04-21T13:21:20+00:00","remoteAddr":"193.250.180.152","user":"mga","app":"webdav","method":"PUT","url":"\/remote.php\/dav\/files\/mga\/MGA\/FACTURATION\/tableau%20facturation.xls","message":"Exception: {\"Message\":\"HTTP\\\/1.1 423 \\\"MGA\\\/FACTURATION\\\/tableau facturation.xls\\\" is locked\",\"Exception\":\"OCA\\\\DAV\\\\Connector\\\\Sabre\\\\Exception\\\\FileLocked\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/owncloud\\\/lib\\\/public\\\/Events\\\/EventEmitterTrait.php(50): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\File->OCA\\\\DAV\\\\Connector\\\\Sabre\\\\{closure}()\\n#1 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/lib\\\/Connector\\\/Sabre\\\/File.php(269): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\File->emittingCall(Object(Closure), Array, 'file', 'create')\\n#2 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(1129): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\File->put(Resource id #84)\\n#3 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/CorePlugin.php(513): Sabre\\\\DAV\\\\Server->updateFile('files\\\/mga\\\/MGA\\\/F...', Resource id #84, NULL)\\n#4 [internal function]: Sabre\\\\DAV\\\\CorePlugin->httpPut(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#5 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n#6 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(479): Sabre\\\\Event\\\\EventEmitter->emit('method:PUT', Array)\\n#7 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(254): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#8 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/lib\\\/Server.php(263): Sabre\\\\DAV\\\\Server->exec()\\n#9 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/appinfo\\\/v2\\\/remote.php(31): OCA\\\\DAV\\\\Server->exec()\\n#10 \\\/var\\\/www\\\/owncloud\\\/remote.php(165): require_once('\\\/var\\\/www\\\/ownclo...')\\n#11 {main}\",\"File\":\"\\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/lib\\\/Connector\\\/Sabre\\\/File.php\",\"Line\":205,\"User\":\"mga\"}"}
{"reqId":"8e7a266e-a32c-4876-be90-1a0df8f53785","level":2,"time":"2018-04-21T13:22:02+00:00","remoteAddr":"109.223.245.190","user":"matthieu","app":"dav","method":"MOVE","url":"\/remote.php\/dav\/uploads\/matthieu\/3386207293\/.file","message":"Could not get node for path: \"uploads\/matthieu\/3386207293\/.file\" : File with name \/\/3386207293 could not be located"}
{"reqId":"04faecb1-8e41-44c9-8aec-59258651d582","level":4,"time":"2018-04-21T14:23:02+00:00","remoteAddr":"109.214.161.203","user":"violaine","app":"webdav","method":"PUT","url":"\/remote.php\/dav\/files\/violaine\/Scan_00099999.pdf","message":"Exception: {\"Message\":\"HTTP\\\/1.1 400 The computed checksum does not match the one received from the client.\",\"Exception\":\"Sabre\\\\DAV\\\\Exception\\\\BadRequest\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/owncloud\\\/lib\\\/public\\\/Events\\\/EventEmitterTrait.php(50): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\File->OCA\\\\DAV\\\\Connector\\\\Sabre\\\\{closure}()\\n#1 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/lib\\\/Connector\\\/Sabre\\\/File.php(269): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\File->emittingCall(Object(Closure), Array, 'file', 'create')\\n#2 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/lib\\\/Connector\\\/Sabre\\\/Directory.php(171): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\File->put(Resource id #84)\\n#3 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(1095): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\Directory->createFile('Scan_00099999.p...', Resource id #84)\\n#4 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/CorePlugin.php(525): Sabre\\\\DAV\\\\Server->createFile('files\\\/violaine\\\/...', Resource id #84, NULL)\\n#5 [internal function]: Sabre\\\\DAV\\\\CorePlugin->httpPut(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#6 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n#7 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(479): Sabre\\\\Event\\\\EventEmitter->emit('method:PUT', Array)\\n#8 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(254): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#9 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/lib\\\/Server.php(263): Sabre\\\\DAV\\\\Server->exec()\\n#10 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/appinfo\\\/v2\\\/remote.php(31): OCA\\\\DAV\\\\Server->exec()\\n#11 \\\/var\\\/www\\\/owncloud\\\/remote.php(165): require_once('\\\/var\\\/www\\\/ownclo...')\\n#12 {main}\",\"File\":\"\\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/lib\\\/Connector\\\/Sabre\\\/File.php\",\"Line\":163,\"User\":\"violaine\"}"}
{"reqId":"04faecb1-8e41-44c9-8aec-59258651d582","level":4,"time":"2018-04-21T14:23:12+00:00","remoteAddr":"109.214.161.203","user":"violaine","app":"webdav","method":"PUT","url":"\/remote.php\/dav\/files\/violaine\/Scan_00099999.pdf","message":"Exception: {\"Message\":\"HTTP\\\/1.1 400 The computed checksum does not match the one received from the client.\",\"Exception\":\"Sabre\\\\DAV\\\\Exception\\\\BadRequest\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/owncloud\\\/lib\\\/public\\\/Events\\\/EventEmitterTrait.php(50): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\File->OCA\\\\DAV\\\\Connector\\\\Sabre\\\\{closure}()\\n#1 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/lib\\\/Connector\\\/Sabre\\\/File.php(269): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\File->emittingCall(Object(Closure), Array, 'file', 'create')\\n#2 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/lib\\\/Connector\\\/Sabre\\\/Directory.php(171): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\File->put(Resource id #84)\\n#3 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(1095): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\Directory->createFile('Scan_00099999.p...', Resource id #84)\\n#4 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/CorePlugin.php(525): Sabre\\\\DAV\\\\Server->createFile('files\\\/violaine\\\/...', Resource id #84, NULL)\\n#5 [internal function]: Sabre\\\\DAV\\\\CorePlugin->httpPut(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#6 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n#7 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(479): Sabre\\\\Event\\\\EventEmitter->emit('method:PUT', Array)\\n#8 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(254): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#9 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/lib\\\/Server.php(263): Sabre\\\\DAV\\\\Server->exec()\\n#10 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/appinfo\\\/v2\\\/remote.php(31): OCA\\\\DAV\\\\Server->exec()\\n#11 \\\/var\\\/www\\\/owncloud\\\/remote.php(165): require_once('\\\/var\\\/www\\\/ownclo...')\\n#12 {main}\",\"File\":\"\\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/lib\\\/Connector\\\/Sabre\\\/File.php\",\"Line\":163,\"User\":\"violaine\"}"}
{"reqId":"91a3a063-c722-498d-9c93-3ff236f5f17f","level":2,"time":"2018-04-21T13:23:40+00:00","remoteAddr":"109.223.245.190","user":"matthieu","app":"dav","method":"DELETE","url":"\/remote.php\/dav\/files\/matthieu\/iorad-capture.exe","message":"Could not get node for path: \"files\/matthieu\/iorad-capture.exe\" : File with name iorad-capture.exe could not be located"}
{"reqId":"QrMrDXR3VSlNh9ewSW0y","level":2,"time":"2018-04-21T13:25:34+00:00","remoteAddr":"193.250.180.152","user":"--","app":"core","method":"PROPFIND","url":"\/remote.php\/dav\/addressbooks\/users\/mga\/contacts\/","message":"Login failed: 'mga' (Remote IP: '193.250.180.152')"}
{"reqId":"3opdwU1fAZBLGfTXbtpE","level":2,"time":"2018-04-21T13:25:34+00:00","remoteAddr":"193.250.180.152","user":"--","app":"core","method":"PROPFIND","url":"\/remote.php\/dav\/calendars\/mga\/personal\/","message":"Login failed: 'mga' (Remote IP: '193.250.180.152')"}

```

#### Browser log
```
nothin !
```

Thx for any help

**Comment by ownclouders** [comment]
[GitMate.io](https://gitmate.io) thinks the contributor most likely able to help you is @PVince81.

Possibly related issues are https://github.com/owncloud/core/issues/11811 (Compute and store file checksums on the fly / file integrity), https://github.com/owncloud/core/issues/28584 (ChunkingNG: Intermittently at MOVE, checksums of individual chunks are computed(?)), https://github.com/owncloud/core/issues/19761 (Unauthorised code doesn't match documentation.), https://github.com/owncloud/core/issues/27920 (Home folder unavailable after upgrade til version 10), and https://github.com/owncloud/core/issues/30751 ([files_antivirus] Upload of Eicar throws Sabre\Exception\UnsupportedMediaType).

**Comment by PVince81** [comment]
Can you manually compare the MD5 / SHA256 checksum of the file as it appears on the data folder with the one from the upload ? Maybe try uploading a text file bigger than 1 MB (ex: log file) and see if you can identify what kind of corruption it is having.

In general this sounds more like an environment issue, something that is interfering with the data stream.

**Comment by Crush33** [comment]
MD5 seems to be ok on each side...
I looked into network parameters of the virtual machine, changed a few things with multiple reboot but the issue is still present for a few people.

It seems that it occurs less often anyway so you must be right when you says that is environment issue...

I'll keep an eye on it today...

@PVince81 : is it possible to give you an ssh access to give a check to our instance if necessary ? Thx !

**Comment by PVince81** [comment]
@Crush33 please see https://owncloud.org/help/ for support as this seems to be an environment issue and not a bug in ownCloud.

**Comment by Crush33** [comment]
After long investigation you were right.
On a graphical debian9 virutalized on Hyper-V server, the network doesn't work well.
I had to 
-apt update && apt upgrade
-install resovlconf package, 
-disable IP settings IPV4 and IPV6 in the network menu in graphic mode
-put the right IP settings in /etc/network/interfaces file
-reboot twice

After this, the upload seems ok for every clients.

Conclusion : Better install owncloud server on debian headless...

**Comment by frazy81** [comment]
I had a similar (likely not exactly the same) issue. I got errors "400 The computed checksum does not match the one received from the client", too. Setup: ownCloud running inside debian guest in VirtualBox, ownCloud Desktop Client running on a debian maschine.

I figured out that the problem was on the client side! The problem and solution was simple: I had a not working, to high, network MTU setting on my client computer (MTU 1500). Solution for me was:

sudo ifconfig eth0 mtu 1492 (exchange eth0 by the network adapter name, make permanent MTU change later if this works out)

that solved the problem for me. I hope to be helping someone else who might be experiencing the same problem.
